packages/volto/src/hooks/user/useFetchUser.js

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import jwtDecode from 'jwt-decode';
+import { getUser } from '@plone/volto/actions';
+
+const useFetchUser = () => {
+  const user = useSelector((state) => state.users?.user);
+  const userId = useSelector((state) =>
+    state.userSession.token ? jwtDecode(state.userSession.token).sub : '',
+  );
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (!user?.id) {
+      dispatch(getUser(userId));
+    }
+  }, [dispatch, userId, user]);
+
+  return user;
+};
+
+export default useFetchUser;
```

Comment:
I'd be inclined to name it `useUser`. The point of refactoring this into a hook is that the code using the hook doesn't really need to be aware of how the user data is found.

Reply:
Done